src/sage/combinat/words/word.py

```diff
@@ -518,10 +518,12 @@ class InfiniteWord_iter_with_caching(WordDatatype_iter_with_caching, InfiniteWor
 
     Pickle is not supported for infinite word defined by an iterator::
 
-        sage: dumps(w)
-        Traceback (most recent call last):
-        ...
-        TypeError: can...t...pickle...generator...object...
+        sage: try:
+        ....:     dumps(w)
+        ....: except TypeError as e:
+        ....:     if "pickle" in str(e) and "generator" in str(e):
+        ....:         print("TypeError raised in dumps() as expected")
+        TypeError raised in dumps() as expected
```

Comment:
Does something like this works?
```suggestion
        sage: try:
        ....:     dumps(w)
        ....: except TypeError as e:
        ....:     print("pickle" in str(e) and "generator" in str(e))
        ....: True
```

Reply:
I follow @dimpase 's suggestions to print some more meaningful words. Actually it also works

Reply:
It's not a big thing, but with your original code one cannot distinguish between the cases where no TypeError is thrown and where a TypeError is thrown but doesn't contain "pickle" or "generator".

Reply:
OK. I know, I will fix this